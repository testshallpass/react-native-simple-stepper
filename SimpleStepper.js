import React, {useState} from 'react';
import {StyleSheet, Text, TouchableOpacity, Image, View} from 'react-native';
const STEP = {
  increment: 'increment',
  decrement: 'decrement',
};
const TEXT_POSITION = {
  left: 'left',
  center: 'center',
  right: 'right',
};

const ImageView = ({
  style = {
    height: 30,
    width: 30,
  },
  source = {},
  opacity = 1,
}) => {
  return <Image style={[style, {opacity}]} source={source} />;
};

const Step = ({
  activeOpacity = 1,
  style = {
    padding: 8,
  },
  onPress = () => {},
  renderImage = undefined,
  disabled = false,
  imageStyle = {
    height: 30,
    width: 30,
  },
  imageOpacity = 1,
  imageSource = undefined,
}) => {
  return (
    <TouchableOpacity
      activeOpacity={activeOpacity}
      style={style}
      onPress={onPress}
      disabled={disabled}>
      {renderImage ? (
        renderImage()
      ) : (
        <ImageView
          style={imageStyle}
          opacity={imageOpacity}
          source={imageSource}
        />
      )}
    </TouchableOpacity>
  );
};

const SimpleStepper = ({
  initialValue = 0,
  minimumValue = 0,
  maximumValue = 10,
  stepValue = 1,
  valueChanged = () => {},
  decrementImage = require('./assets/decrement.png'),
  incrementImage = require('./assets/increment.png'),
  activeOpacity = 0.4,
  disabledOpacity = 0.5,
  disabled = false,
  renderDecrementImage = undefined,
  renderIncrementImage = undefined,
  renderDecrementStep = undefined,
  renderIncrementStep = undefined,
  wraps = false,
  onMin = () => {},
  onMax = () => {},
  onIncrement = () => {},
  onDecrement = () => {},
  showText = false,
  renderText = undefined,
  textStyle = {
    padding: 4,
    fontSize: 20,
    fontWeight: 'bold',
    color: 'blue',
  },
  containerStyle = {
    backgroundColor: 'transparent',
    flexDirection: 'row',
    borderWidth: 1,
    borderRadius: 8,
    overflow: 'hidden',
    alignItems: 'center',
    borderColor: 'blue',
  },
  separatorStyle = {
    width: StyleSheet.hairlineWidth,
    backgroundColor: 'blue',
    height: '100%',
  },
  incrementStepStyle = {
    padding: 4,
  },
  decrementStepStyle = {
    padding: 4,
  },
  incrementImageStyle = {
    height: 30,
    width: 30,
  },
  decrementImageStyle = {
    height: 30,
    width: 30,
  },
  textPosition = TEXT_POSITION.center,
}) => {
  const [value, setValue] = useState(initialValue);

  const _decrementAction = () => {
    const nextValue = value - stepValue;
    _processValue(nextValue, true, onDecrement);
  };

  const _incrementAction = () => {
    const nextValue = value + stepValue;
    _processValue(nextValue, true, onIncrement);
  };

  const _processValue = (
    newValue = 0,
    changed = false,
    onAction = () => {},
  ) => {
    const {hasReachedMin, hasReachedMax} = _getHasMinMax(newValue);
    if (newValue > maximumValue) {
      newValue = wraps ? minimumValue : maximumValue;
    } else if (newValue === maximumValue) {
      newValue = maximumValue;
    } else if (newValue < minimumValue) {
      newValue = wraps ? maximumValue : minimumValue;
    } else if (newValue === minimumValue) {
      newValue = minimumValue;
    }
    onAction(newValue);
    if (hasReachedMin) {
      onMin(newValue);
    }
    if (hasReachedMax) {
      onMax(newValue);
    }
    if (changed) {
      valueChanged(newValue);
      setValue(newValue);
    }
  };

  const _getHasMinMax = (newValue = 0) => {
    let hasReachedMax = true;
    let hasReachedMin = true;
    switch (true) {
      case wraps:
        hasReachedMin = false;
        hasReachedMax = false;
        break;
      case stepValue >= 0:
        hasReachedMax = newValue >= maximumValue;
        hasReachedMin = newValue <= minimumValue;
        break;
      case stepValue < 0:
        // step value is negative
        // swap the max and min conditions.
        hasReachedMax = newValue <= minimumValue;
        hasReachedMin = newValue >= maximumValue;
        break;
    }
    return {
      hasReachedMax,
      hasReachedMin,
    };
  };

  const _getImageSource = (type, source) => {
    if (typeof source === 'string') {
      if (source.length === 0) {
        if (type === STEP.decrement) {
          return require('./assets/decrement.png');
        } else if (type === STEP.increment) {
          return require('./assets/increment.png');
        }
      } else {
        return {uri: source};
      }
    }
    return source;
  };

  const _getImageViewProps = (type, opacity) => {
    let style = {
      height: 30,
      width: 30,
    };
    let imageSource = null;
    switch (type) {
      case STEP.increment:
        style = incrementImageStyle;
        imageSource = incrementImage;
        break;
      case STEP.decrement:
        style = decrementImageStyle;
        imageSource = decrementImage;
        break;
    }
    return {
      style,
      opacity,
      source: _getImageSource(type, imageSource),
    };
  };

  const _renderText = newValue => {
    if (renderText) {
      return renderText(newValue);
    }
    return <Text style={textStyle}>{newValue}</Text>;
  };

  const {hasReachedMin, hasReachedMax} = _getHasMinMax(value);
  const decrementOpacity = hasReachedMin || disabled ? disabledOpacity : 1;
  const incrementOpacity = hasReachedMax || disabled ? disabledOpacity : 1;
  const decrementImageProps = _getImageViewProps(
    STEP.decrement,
    decrementOpacity,
  );
  const incrementImageProps = _getImageViewProps(
    STEP.increment,
    incrementOpacity,
  );
  const isLeft = showText && textPosition === TEXT_POSITION.left;
  const isCenter = showText && textPosition === TEXT_POSITION.center;
  const isRight = showText && textPosition === TEXT_POSITION.right;
  return (
    <View>
      <View style={containerStyle}>
        {isLeft && _renderText(value, renderText, textStyle)}
        {isLeft && <View style={separatorStyle} />}
        {renderDecrementStep ? (
          renderDecrementStep()
        ) : (
          <Step
            style={decrementStepStyle}
            activeOpacity={activeOpacity}
            onPress={_decrementAction}
            disabled={hasReachedMin || disabled}
            renderImage={renderIncrementImage}
            imageStyle={decrementImageProps.style}
            imageOpacity={decrementImageProps.opacity}
            imageSource={decrementImageProps.source}
          />
        )}
        {isCenter && <View style={separatorStyle} />}
        {isCenter && _renderText(value)}
        <View style={separatorStyle} />
        {renderIncrementStep ? (
          renderIncrementStep()
        ) : (
          <Step
            style={incrementStepStyle}
            activeOpacity={activeOpacity}
            onPress={_incrementAction}
            disabled={hasReachedMax || disabled}
            renderImage={renderDecrementImage}
            imageStyle={incrementImageProps.style}
            imageOpacity={incrementImageProps.opacity}
            imageSource={incrementImageProps.source}
          />
        )}
        {isRight && <View style={separatorStyle} />}
        {isRight && _renderText(value)}
      </View>
    </View>
  );
};

export default SimpleStepper;
