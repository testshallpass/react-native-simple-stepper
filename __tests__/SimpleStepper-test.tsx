import React from 'react';
import {TouchableOpacity, Text, Image} from 'react-native';
import '@testing-library/jest-native/extend-expect';
import {
  render,
  cleanup,
  screen,
  fireEvent,
} from '@testing-library/react-native';
import SimpleStepper from '../SimpleStepper';

afterEach(cleanup);

test('it renders', () => {
  const component = render(<SimpleStepper />);
  expect(component).toBeDefined();
  expect(component.toJSON()).toMatchSnapshot();
});

test('increment by 1 and decrement by 1', async () => {
  let actualValue = 0;
  render(<SimpleStepper valueChanged={value => (actualValue = value)} />);

  const decrementButton = await screen.findByTestId('decrementButton');
  expect(decrementButton).toBeDefined();

  const incrementButton = await screen.findByTestId('incrementButton');
  expect(incrementButton).toBeDefined();

  fireEvent.press(incrementButton);
  expect(actualValue).toBe(1);

  fireEvent.press(decrementButton);
  expect(actualValue).toBe(0);
});

test('increment by 1 until hit maximum value of 3', async () => {
  let actualValue = 0;
  const max = 3;
  render(
    <SimpleStepper
      maximumValue={max}
      valueChanged={value => (actualValue = value)}
    />,
  );

  const incrementButton = await screen.findByTestId('incrementButton');
  expect(incrementButton).toBeDefined();

  for (let index = 1; index <= max; index++) {
    fireEvent.press(incrementButton);
    expect(actualValue).toBe(index);
  }
  expect(actualValue).toBe(max);

  const decrementButton = await screen.findByTestId('incrementButton');
  expect(decrementButton).toBeDefined();

  for (let index = 2; index < 0; index--) {
    fireEvent.press(decrementButton);
    expect(actualValue).toBe(index);
  }
});

test('it wraps', async () => {
  let actualValue = 0;
  const max = 3;
  render(
    <SimpleStepper
      wraps
      maximumValue={max}
      valueChanged={value => (actualValue = value)}
    />,
  );

  const incrementButton = await screen.findByTestId('incrementButton');
  expect(incrementButton).toBeDefined();

  let expected = [1, 2, 3, 0, 1, 2, 3, 0, 1, 2, 3, 0];

  for (let index = 0; index < expected.length; index++) {
    const value = expected[index];
    fireEvent.press(incrementButton);
    expect(actualValue).toBe(value);
  }

  expect(actualValue).toBe(0);

  const decrementButton = await screen.findByTestId('decrementButton');
  expect(decrementButton).toBeDefined();

  expected = [3, 2, 1, 0, 3, 2, 1, 0, 3, 2, 1, 0];

  for (let index = 0; index < expected.length; index++) {
    const value = expected[index];
    fireEvent.press(decrementButton);
    expect(actualValue).toBe(value);
  }
});

test('show center text', () => {
  const component = render(<SimpleStepper showText />);
  expect(component).toBeDefined();
  expect(component.toJSON()).toMatchSnapshot();

  const text = screen.getByText('0');
  expect(text).toBeDefined();
});

test('useColor and color', async () => {
  render(<SimpleStepper showText useColor color={'red'} />);

  const text = screen.getByText('0');
  expect(text).toBeDefined();
  expect(text).toHaveStyle({
    marginHorizontal: 8,
    fontSize: 24,
    color: 'red',
  });

  const container = await screen.findByTestId('container');
  expect(container).toBeDefined();
  expect(container).toHaveStyle({
    flexDirection: 'row',
    borderWidth: 1,
    borderRadius: 8,
    alignItems: 'center',
    justifyContent: 'space-evenly',
    borderColor: 'red',
  });

  const separator = await screen.findByTestId('separator');
  expect(separator).toBeDefined();
  expect(separator).toHaveStyle({
    width: 0.5,
    backgroundColor: 'red',
    height: '100%',
  });
});

test('render text', () => {
  const component = render(
    <SimpleStepper showText renderText={value => <Text>{value}</Text>} />,
  );
  expect(component).toBeDefined();
  expect(component.toJSON()).toMatchSnapshot();

  const text = screen.getByText('0');
  expect(text).toBeDefined();
});

test('show decimal text', () => {
  render(<SimpleStepper showText initialValue={0.1337} />);
  const text = screen.getByText('0.13');
  expect(text).toBeDefined();
});

test('negative step value', async () => {
  let actualValue = 0;
  render(
    <SimpleStepper
      stepValue={-1}
      valueChanged={value => (actualValue = value)}
    />,
  );

  const decrementButton = await screen.findByTestId('decrementButton');
  expect(decrementButton).toBeDefined();

  const incrementButton = await screen.findByTestId('incrementButton');
  expect(incrementButton).toBeDefined();

  fireEvent.press(incrementButton);
  expect(actualValue).toBe(0);

  fireEvent.press(decrementButton);
  expect(actualValue).toBe(1);
});

test('disable tint color', async () => {
  render(
    <SimpleStepper
      incrementImageStyle={{height: 30, width: 30, tintColor: 'blue'}}
      decrementImageStyle={{height: 30, width: 30, tintColor: 'blue'}}
      disableIncrementImageTintColor
      disableDecrementImageTintColor
    />,
  );
  const decrementImage = await screen.findByTestId('decrementImage');
  expect(decrementImage).toBeDefined();
  expect(decrementImage).toHaveStyle({
    height: 30,
    width: 30,
    tintColor: undefined,
  });

  const incrementImage = await screen.findByTestId('incrementImage');
  expect(incrementImage).toBeDefined();
  expect(incrementImage).toHaveStyle({
    height: 30,
    width: 30,
    tintColor: undefined,
  });
});

test('render increment and decrement', () => {
  const reactNativeFavicon = {
    uri: 'https://reactnative.dev/docs/assets/favicon.png',
  };
  const component = render(
    <SimpleStepper
      renderIncrementImage={opacity => (
        <Image source={reactNativeFavicon} style={{opacity}} />
      )}
      renderDecrementImage={opacity => (
        <Image source={reactNativeFavicon} style={{opacity}} />
      )}
    />,
  );
  expect(component).toBeDefined();
  expect(component.toJSON()).toMatchSnapshot();
});

test('render increment step and decrement step', async () => {
  const component = render(
    <SimpleStepper
      maximumValue={3}
      renderIncrementStep={(value, onIncrement) => (
        <TouchableOpacity testID={'incrementBtn'} onPress={onIncrement}>
          <Text>{value}</Text>
        </TouchableOpacity>
      )}
      renderDecrementStep={(value, onDecrement) => (
        <TouchableOpacity testID={'decrementBtn'} onPress={onDecrement}>
          <Text>{value}</Text>
        </TouchableOpacity>
      )}
    />,
  );
  expect(component).toBeDefined();
  expect(component.toJSON()).toMatchSnapshot();

  const decrementButton = await screen.findByTestId('decrementBtn');
  expect(decrementButton).toBeDefined();
  fireEvent.press(decrementButton);

  const incrementButton = await screen.findByTestId('incrementBtn');
  expect(incrementButton).toBeDefined();

  for (let index = 0; index < 4; index++) {
    fireEvent.press(incrementButton);
  }
});

test('show left text', () => {
  const component = render(<SimpleStepper showText textPosition={'left'} />);
  expect(component).toBeDefined();
  expect(component.toJSON()).toMatchSnapshot();

  const text = screen.getByText('0');
  expect(text).toBeDefined();
});

test('show right text', () => {
  const component = render(<SimpleStepper showText textPosition={'right'} />);
  expect(component).toBeDefined();
  expect(component.toJSON()).toMatchSnapshot();

  const text = screen.getByText('0');
  expect(text).toBeDefined();
});
