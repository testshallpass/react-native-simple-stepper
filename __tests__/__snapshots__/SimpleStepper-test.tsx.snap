// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="decrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/decrement.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 0.5,
            },
          ]
        }
        testID="decrementImage"
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="incrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/increment.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 1,
            },
          ]
        }
        testID="incrementImage"
      />
    </View>
  </View>
</View>
`;

exports[`render increment and decrement 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="decrementButton"
    >
      <Image
        source={
          {
            "uri": "https://reactnative.dev/docs/assets/favicon.png",
          }
        }
        style={
          {
            "opacity": 0.5,
          }
        }
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="incrementButton"
    >
      <Image
        source={
          {
            "uri": "https://reactnative.dev/docs/assets/favicon.png",
          }
        }
        style={
          {
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`render increment step and decrement step 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="decrementBtn"
    >
      <Text>
        0
      </Text>
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="incrementBtn"
    >
      <Text>
        0
      </Text>
    </View>
  </View>
</View>
`;

exports[`render text 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="decrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/decrement.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 0.5,
            },
          ]
        }
        testID="decrementImage"
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
      testID="separator"
    />
    <Text>
      0
    </Text>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="incrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/increment.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 1,
            },
          ]
        }
        testID="incrementImage"
      />
    </View>
  </View>
</View>
`;

exports[`show center text 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="decrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/decrement.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 0.5,
            },
          ]
        }
        testID="decrementImage"
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
      testID="separator"
    />
    <Text
      style={
        {
          "fontSize": 24,
          "marginHorizontal": 8,
        }
      }
    >
      0
    </Text>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="incrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/increment.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 1,
            },
          ]
        }
        testID="incrementImage"
      />
    </View>
  </View>
</View>
`;

exports[`show left text 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="container"
  >
    <Text
      style={
        {
          "fontSize": 24,
          "marginHorizontal": 8,
        }
      }
    >
      0
    </Text>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
      testID="separator"
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="decrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/decrement.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 0.5,
            },
          ]
        }
        testID="decrementImage"
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="incrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/increment.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 1,
            },
          ]
        }
        testID="incrementImage"
      />
    </View>
  </View>
</View>
`;

exports[`show right text 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
      }
    }
    testID="container"
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": true,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="decrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/decrement.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 0.5,
            },
          ]
        }
        testID="decrementImage"
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "padding": 4,
        }
      }
      testID="incrementButton"
    >
      <Image
        source={
          {
            "testUri": "../../../assets/increment.png",
          }
        }
        style={
          [
            {
              "height": 30,
              "width": 30,
            },
            {
              "opacity": 1,
            },
          ]
        }
        testID="incrementImage"
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "black",
          "height": "100%",
          "width": 0.5,
        }
      }
      testID="separator"
    />
    <Text
      style={
        {
          "fontSize": 24,
          "marginHorizontal": 8,
        }
      }
    >
      0
    </Text>
  </View>
</View>
`;
